posthog/temporal/common/logger.py

```diff
@@ -56,22 +59,99 @@ async def bind_temporal_org_worker_logger(
     temporal_context = get_temporal_context()
 
     return logger.new(organization_id=str(organization_id), destination=destination, **temporal_context)
 
 
-def configure_logger(
+def configure_logger_sync(
+    logger_factory=structlog.PrintLoggerFactory,
+    extra_processors: list[structlog.types.Processor] | None = None,
+    queue: sync_queue.Queue | None = None,
+    producer: KafkaProducer | None = None,
+    cache_logger_on_first_use: bool = True,
+) -> None:
+    """Configure a sync StructLog logger for temporal workflows.
+
+    Keep up to date with the async version `configure_logger_async`
+
+    Configuring the logger involves:
+    * Setting up processors.
+    * Spawning a task to listen for Kafka logs.
+    * Spawning a task to shutdown gracefully on worker shutdown.
+
+    Args:
+        logger_factory: Optionally, override the logger_factory.
+        extra_processors: Optionally, add any processors at the end of the chain.
+        queue: Optionally, bring your own log queue.
+        producer: Optionally, bring your own Kafka producer.
+        cache_logger_on_first_use: Set whether to cache logger for performance.
+            Should always be True except in tests.
+    """
+
+    base_processors: list[structlog.types.Processor] = [
+        structlog.processors.add_log_level,
+        structlog.processors.format_exc_info,
+        structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f", utc=True),
+        structlog.stdlib.PositionalArgumentsFormatter(),
+    ]
+
+    log_queue = queue if queue is not None else sync_queue.Queue(maxsize=-1)
+    log_producer = None
+    log_producer_error = None
+
+    try:
+        log_producer = KafkaLogProducerFromQueueSync(queue=log_queue, topic=KAFKA_LOG_ENTRIES, producer=producer)
+    except Exception as e:
+        # Skip putting logs in queue if we don't have a producer that can consume the queue.
+        # We save the error to log it later as the logger hasn't yet been configured at this time.
+        log_producer_error = e
+    else:
+        put_in_queue = PutInLogQueueProcessor(log_queue)
+        base_processors.append(put_in_queue)
+
+    base_processors += [
+        EventRenamer("msg"),
+        structlog.processors.JSONRenderer(),
+    ]
+    extra_processors_to_add = extra_processors if extra_processors is not None else []
+
+    structlog.configure(
+        processors=base_processors + extra_processors_to_add,
+        logger_factory=logger_factory(),
+        cache_logger_on_first_use=cache_logger_on_first_use,
+    )
+
+    if log_producer is None:
+        logger = structlog.get_logger()
+        logger.error("Failed to initialize log producer", exc_info=log_producer_error)
+        return
+
+    listener_thread = threading.Thread(target=log_producer.listen, daemon=True)
+    listener_thread.start()
+
+    def worker_shutdown_handler():
+        temporalio.activity.wait_for_worker_shutdown_sync()
+        log_queue.join()
+        listener_thread.join()
```

Comment:
`join` will not terminate the underlying thread, but block until they do because they reach the end of their execution or an unhandled exception is raised. None of these two will happen, so these `join`s here will wait forever.

Reply:
Looks like we do have a stop signal if the queue gets a `None` message, but I don't see that implemented? Maybe I'm missing something.

Reply:
There's a stop signal thing on line 447 - maybe that's all we need?

Reply:
I think something like this:
```suggestion
        log_queue.join()
        log_queue.put(None)
        listener_thread.join()
```

Maybe works?

Reply:
Fixed